welearn_datastack/modules/pdf_extractor.py

```diff
@@ -1,49 +1,69 @@
+import io
 import logging
-from typing import List, Tuple
+from typing import List
 
-from pypdf import PdfReader
+from bs4 import BeautifulSoup
 from refinedoc.refined_document import RefinedDocument
 
+from welearn_datastack.utils_.http_client_utils import get_new_https_session
+
 logger = logging.getLogger(__name__)
 
 
-def large_pages_size_flag(reader: PdfReader, limit: int) -> Tuple[List[int], bool]:
+def _send_pdf_to_tika(pdf_content: io.BytesIO, tika_base_url: str) -> dict:
+    """
+    Send a PDF document to Tika micro service and return the content as a dictionary
+    :param pdf_content: the PDF document content as a byte stream
+    :param tika_base_url: the base URL of the Tika micro service
+    :return: the content returned by Tika micro service as a dictionary (JSON)
+    """
+    if tika_base_url.endswith("/"):
+        tika_base_url = tika_base_url[:-1]
```

Comment:
:bulb: to avoid the condition
```suggestion
    tika_base_url = tutu = re.sub(r"\/$", "", tika_base_url)
```